apps/chat/src/store/import-export/importExport.epics.ts

```diff
@@ -392,75 +409,105 @@ const uploadConversationAttachmentsEpic: AppEpic = (action$) =>
-                id,
-                name,
-                absolutePath,
-                relativePath,
-                status,
-                percent,
-                contentType,
-              } = result;
-              return ImportExportActions.uploadSingleAttachmentSuccess({
-                apiResult: {
-                  id,
-                  name,
-                  absolutePath,
-                  relativePath,
-                  status,
-                  percent,
-                  contentType,
-                },
-              });
+          const rootFolderName = isFolderNameExist
+            ? generateNextName(conversationName, conversationName, folders)
+            : conversationName;
+
+          const actions = attachmentsToUpload.map((attachment) => {
+            const formData = new FormData();
+
+            if (!attachment.fileContent) {
+              return of(
+                ImportExportActions.uploadSingleFileFail({
+                  id: attachment.id,
+                }),
+              );
             }
 
-            return FilesActions.uploadFileTick({
-              id: attachment.id,
-              percent: percent!,
-            });
-          }),
-          catchError(() => {
-            return of(
-              ImportExportActions.uploadSingleFileFail({
-                id: attachment.id,
+            formData.append(
+              'attachment',
+              attachment.fileContent,
+              attachment.name,
+            );
+            const { parentPath } = splitEntityId(attachment.id);
+            const newParentPath =
+              parentPath && parentPath.replace(`${ImportRoot.Imports}/`, '');
+
+            const relativePath = constructPath(
+              ImportRoot.Imports,
+              rootFolderName,
+              newParentPath,
+            );
+
+            return FileService.sendFile(
+              formData,
+              relativePath,
+              attachment.name,
+            ).pipe(
+              filter(
+                ({ percent, result }) =>
+                  typeof percent !== 'undefined' ||
+                  typeof result !== 'undefined',
+              ),
+              map(({ percent, result }) => {
+                if (result) {
+                  const {
+                    id,
+                    name,
+                    absolutePath,
+                    relativePath,
+                    status,
+                    percent,
+                    contentType,
+                  } = result;
+                  return ImportExportActions.uploadSingleAttachmentSuccess({
+                    apiResult: {
+                      id,
+                      name,
+                      absolutePath,
+                      relativePath,
+                      status,
+                      percent,
+                      contentType,
+                      oldId: attachment.id,
+                    },
+                  });
+                }
```

Comment:
```suggestion
              map(({ percent, result }) => {
                if (result) {
                  return ImportExportActions.uploadSingleAttachmentSuccess({
                    apiResult: {
                      ...result,
                      oldId: attachment.id,
                    },
                  });
                }
```

Reply:
Thank you for the suggestion!
Redundant object destruction removed.